main/src/main.ts

```diff
@@ -242,6 +242,11 @@ export function createWindow(
         Log.info("...closed event completed")
     })
 
+    currentWindow.webContents.on("will-navigate", (event, url) => {
+        event.preventDefault()
+        currentWindow.webContents.send("open-oni-browser", url)
```

Comment:
One minor thing - we may not actually need a new `ipc` message in this case. I think we could use the `execute-command` one we have, along with the `browser.openUrl` action, something like:
currentWindow.webContents.send("execute-command", "browser.openUrl", url)

We do something similar in the menu:
https://github.com/onivim/oni/blob/16c2eb508a5c69e4cc1322bd6395a02065ca5912/main/src/menu.ts#L50